CleverRaven/Cataclysm-DDA
# Fix #76414: Can't activate item - character instead tries to eat it.

#### Summary
Bugfixes "Fixes #76414, Can't activate item - character instead tries to eat it."

#### Purpose of change
Fixes #76414 .

Steps to reproduce:
1. Try to make some lard in a batch.
2. Stop the action.
3. Attempt to resume it
4. Fail bc your character tries to eat it.

Can be duplicated by trying to resume any comestible craft.  Also works with trying to activate pet foods such as cattle fodder.  

#### Describe the solution
Revert commit e0650e55d3ec5e195738e83d71ba0b02b5303578 .  This commit changed the priority order by which item methods gets called, specifically it puts the consume action ahead of the use action in the priority queue.  By reverting this commit the priority queue returns to the original order of use then consume.  

A consequence / trade-off of this PR is that Comestible items that also have a use action defined to them such as delayed transform will no longer be able to be consumed.   Having reviewed the comestible items that also have a use action, I beleive that the consumption of delayed transformation items are a rare edge case that most players would never try.  These are disgusting items barely fit for consumption akin to dirty pool water or food soaked in brine.  I think a higher priority demand for the players is to enable the main line use of these items which is to ferment or to feed pets and that sacrificing our ability to drink dirty pool water is worth it to make sure that Fido doesn't starve.  The functionality to drink dirty pool water can be restored at a later lower priority PR. 

#### Describe alternatives you've considered
Altering the delayed transform code to give the player the choice on comestibles as to whether to activate the transformation or to consume the item.  The level of effort for this would be much higher and meanwhile my Fido is starving.

#### Testing

1. Build and launch the game
2. Open the debug menu and spawn Cattle Fodder
3. Confirm that the pet food dialog box opens up.
4. Spawn purifying water in the debug menu
5. Activate the purifying water.
6. Verifying the message, "The chlorine released by the tablets is still working.  You still need to wait a bit before you can drink it." appears in the log.
7. Run Unit Test Suite

#### Additional context
The Master Branch is also giving unit test failures in Russian Translation.

![image](https://github.com/user-attachments/assets/83596ec9-07ee-4165-bb99-6a7b7ea3f047)

![image](https://github.com/user-attachments/assets/b7a72c81-c592-4807-93a7-b0215c2cf062)

![image](https://github.com/user-attachments/assets/2937adb4-f359-46cb-97db-e67c7abbd7bd)
